samples/Basic_Blink/Makefile

```diff
@@ -1,24 +1,42 @@
-#####################################################################
-#### Please don't change this file. Use Makefile-user.mk instead ####
-#####################################################################
-# Including user Makefile.
-# Should be used to set project-specific parameters
-include ./Makefile-user.mk
-
-# Important parameters check.
-# We need to make sure SMING_HOME and ESP_HOME variables are set.
-# You can use Makefile-user.mk in each project or use enviromental variables to set it globally.
- 
+#	Makefile
+#
+#==============================================================================
+#	project settings 
+#	see sming.mk for more settings
+#------------------------------------------------------------------------------
+#	flash size (required)
+SPI_SIZE        = 4M
+
+#	your source directories (default: app)
+MODULES         = app 
+
+#	your extra include directies (default: none)
+EXTRA_INCDIR    = 
+
+#	your extra libraries (default: none)
+EXTRA_LIBS      = 
+
+#	spiff size (default: 0)
+SPIFF_SIZE		= 0
+
+#	reserved flash area for user data (default: 0)
+USER_FLASH_SIZE	= 0
+
+#	patch init data for system_get_vdd33 (default: 0)
+GET_VDD33		= 0
+
+#==============================================================================
+#	now let sming do all the work ...
+#------------------------------------------------------------------------------
+#	make sure that SMING_HOME is set and the path compatible with make 
 ifndef SMING_HOME
-$(error SMING_HOME is not set. Please configure it in Makefile-user.mk)
-endif
-ifndef ESP_HOME
```

Comment:
IMHO it is a good idea not to have settings regarding the build enviroment withing the project files. That is why I think it is better to set _SMING_HOME_ as as a enviroment variable. The makefile than can find and include _sming.mk_ and all plattform specific settings (e. g. _windows.mk_).